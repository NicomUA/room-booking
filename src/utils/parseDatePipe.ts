import { PipeTransform, Injectable, BadRequestException } from '@nestjs/common';

@Injectable()
export class ParseDatePipe implements PipeTransform {
  transform(value: any) {
    if (!value) {
      throw new BadRequestException('Date should be provided');
    }
    // ISO Date regular expression check
    if (!/\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3}Z/.test(value)) {
      throw new BadRequestException(
        'Invalid date format. Expected YYYY-MM-DDTHH:MM:SS.MMMZ',
      );
    }

    return new Date(value);
  }
}
