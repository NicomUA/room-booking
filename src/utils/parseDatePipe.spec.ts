import { ParseDatePipe } from './parseDatePipe';

describe('parseDatePipe unit test', () => {
  let pipe: ParseDatePipe;

  beforeAll(() => {
    pipe = new ParseDatePipe();
  });

  it('should return date', () => {
    expect(pipe.transform('2020-01-01T12:30:00.000Z')).toEqual(
      new Date('2020-01-01T12:30:00.000Z'),
    );
  });

  it('should throw an error if date format is wrong', async () => {
    const result = async () => await pipe.transform('test');
    expect(result).rejects.toThrowErrorMatchingSnapshot();
  });

  it('should throw an error if value not provided', async () => {
    const result = async () => await pipe.transform(null);
    expect(result).rejects.toThrowErrorMatchingSnapshot();
  });
});
