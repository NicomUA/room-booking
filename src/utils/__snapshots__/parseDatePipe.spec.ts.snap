// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseDatePipe unit test should throw an error if date format is wrong 1`] = `"Invalid date format. Expected YYYY-MM-DDTHH:MM:SS.MMMZ"`;

exports[`parseDatePipe unit test should throw an error if value not provided 1`] = `"Date should be provided"`;
